src/components/vive-controls.js

```diff
@@ -43,36 +47,97 @@ module.exports.Component = registerComponent('vive-controls', {
+    this.everGotGamepadEvent = false;
+    this.lastControllerCheck = 0;
   },
 
-  play: function () {
+  startListening: function () {
     var el = this.el;
     el.addEventListener('buttonchanged', this.onButtonChanged);
     el.addEventListener('buttondown', this.onButtonDown);
     el.addEventListener('buttonup', this.onButtonUp);
     el.addEventListener('model-loaded', this.onModelLoaded);
   },
 
-  pause: function () {
+  stopListening: function () {
     var el = this.el;
     el.removeEventListener('buttonchanged', this.onButtonChanged);
     el.removeEventListener('buttondown', this.onButtonDown);
     el.removeEventListener('buttonup', this.onButtonUp);
     el.removeEventListener('model-loaded', this.onModelLoaded);
   },
 
-  update: function () {
+  checkIfControllerPresent: function () {
+    var data = this.data;
+    var isPresent = false;
+    var controller = data.hand === 'right' ? 0 : data.hand === 'left' ? 1 : 2;
+    var numopenvr = 0;
+    trackedControlsUtils.enumerateGamepads(function (gamepad) {
+      if (numopenvr === controller) {
+        isPresent = true;
+      }
+      numopenvr++;
+    }, this.idPrefix);
+
+    if (isPresent !== this.controllerPresent) {
+      this.controllerPresent = isPresent;
+      if (isPresent) {
+        this.injectTrackedControls(); // inject track-controls
+        this.startListening();
+      } else {
+        this.stopListening();
+      }
+    }
+  },
+
+  onGamepadConnected: function (evt) {
+    this.everGotGamepadEvent = true;
+    this.checkIfControllerPresent();
+  },
+
+  onGamepadDisconnected: function (evt) {
+    this.everGotGamepadEvent = true;
+    this.checkIfControllerPresent();
+  },
+
+  play: function () {
+    this.checkIfControllerPresent();
+    window.addEventListener('gamepadconnected', this.onGamepadConnected, false);
+    window.addEventListener('gamepaddisconnected', this.onGamepadDisconnected, false);
+  },
+
+  pause: function () {
+    window.removeEventListener('gamepadconnected', this.onGamepadConnected, false);
+    window.removeEventListener('gamepaddisconnected', this.onGamepadDisconnected, false);
+    this.stopListening();
+  },
+
+  injectTrackedControls: function () {
     var el = this.el;
     var data = this.data;
     var objUrl = 'url(' + VIVE_CONTROLLER_MODEL_OBJ_URL + ')';
     var mtlUrl = 'url(' + VIVE_CONTROLLER_MODEL_OBJ_MTL + ')';
-    // handId: 0 - right, 1 - left
-    var controller = data.hand === 'right' ? 0 : 1;
-    el.setAttribute('tracked-controls', 'controller', controller);
+
+    // handId: 0 - right, 1 - left, 2 - anything else...
+    var controller = data.hand === 'right' ? 0 : data.hand === 'left' ? 1 : 2;
+    // if we have an OpenVR Gamepad, use the fixed mapping
+    el.setAttribute('tracked-controls', {id: data.idPrefix, controller: controller, rotationOffset: data.rotationOffset});
+
     if (!data.model) { return; }
     el.setAttribute('obj-model', {obj: objUrl, mtl: mtlUrl});
   },
 
+  tick: function () {
```

Comment:
from a closer look at https://www.w3.org/TR/gamepad/ there is no specified method to confirm gamepad identity between enumerations -- we already see that Vive has multiple controllers with the id `OpenVR Gamepad` - so don't have basis for emulating `gamepadconnected` / `gamepaddisconnected`.  will take a look at emitting an event from system `tracked-controls` tick instead of making another one